### PR TITLE
Setup cob4-5

### PR DESCRIPTION
**Pull request by ipa-cob4-5**

### CODE REVIEW

**Comment by ipa-cob4-5** [comment]
The setup is still not complete, only the base components are tested.

**Comment by floweisshardt** [comment]
@ipa-nhg: travis is failing due to roslaunch check failure. please help travis to get happy for this PR.

**Comment by NadiaHG** [comment]
I think that the `_ctest_cob_bringup_roslaunch-check_robots_cob4-5.xml` fails because the cob_calibration_data pull request is still not merged: https://github.com/ipa320/cob_calibration_data/pull/100 , and the file `cob_calibration_data/cob4-5/calibration/calibration_offset.urdf.xacro` is required to load the robot description.

Sorry, my bad,  I forgot the comment :  This pull request requires https://github.com/ipa320/cob_calibration_data/pull/100

**Comment by floweisshardt** [comment]
just restarted the Travis job as https://github.com/ipa320/cob_calibration_data/pull/100 is merged now: see https://travis-ci.org/ipa320/cob_robots/builds/125587032

**Comment by floweisshardt** [comment]
looks good now

**Comment by NadiaHG** [comment]
if https://github.com/ipa320/cob_robots/pull/476 and https://github.com/ipa320/cob_robots/pull/477/files are merged, the pull request https://github.com/ipa-cob4-5/cob_robots/pull/1 have to be accepted

File: cob_default_robot_config/cob4-5/torso_joint_configurations.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+joint_names: ["torso_2_joint","torso_3_joint"]
```
Comment (NadiaHG) [inline cob_default_robot_config/cob4-5/torso_joint_configurations.yaml]: cob4-5 hasn't got a torso

File: cob_hardware_config/cob4-5/cob4-5.rviz [hunk 2]
```diff
@@ -0,0 +1 @@
+../common/cob4.rviz
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/cob4-5.rviz]: have to be updated for cob4-5

File: cob_hardware_config/cob4-5/config/arm_left_driver.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+sync:
+  interval_ms: 10
+  overflow: 0
+
+defaults:
+  eds_pkg: cob_hardware_config
+  eds_file: "cob4-2/config/Schunk_0_63.dcf"
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/arm_left_driver.yaml]: use a specific cob4-5 file or a common file, not cob4-2/config/...

File: cob_hardware_config/cob4-5/config/head_cartesian_controller.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+## Cartesian controller parameters
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/head_cartesian_controller.yaml:1]: cob4-5 has not head, this file is not needed

File: cob_hardware_config/cob4-5/config/head_controller.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#all
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/head_controller.yaml:1]: cob4-5 has not head, file can be removed

File: cob_hardware_config/cob4-5/config/head_self_collision.yaml [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/head_self_collision.yaml:1]: remove file

File: cob_hardware_config/cob4-5/config/phidgets.yaml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+frequency: 30                       #nodes update frequency in hz
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/phidgets.yaml]: cob4-5 has not phidgets

File: cob_hardware_config/cob4-5/config/teleop.yaml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    twist_topic_name: '/arm_left/twist_controller/command_twist',
+    twist_max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3]
+  },
+  arm_right: {
+    sss_default_target: "folded",
+    velocity_topic_name: "/arm_right/joint_group_velocity_controller/command",
+    joint_velocity: [0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05],
+    twist_topic_name: '/arm_right/twist_controller/command_twist',
+    twist_max_velocity: [0.3, 0.2, 0.3, 0.3, 0.2, 0.3]
+  }
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/teleop.yaml:88]: also init grippers?

File: cob_hardware_config/cob4-5/config/torso_cartesian_controller.yaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
+## Cartesian controller parameters
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/torso_cartesian_controller.yaml]: remove file

File: cob_hardware_config/cob4-5/config/torso_controller.yaml [hunk 10]
```diff
@@ -0,0 +1,62 @@
+#all
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/torso_controller.yaml]: remove file

File: cob_hardware_config/cob4-5/config/torso_driver.yaml [hunk 11]
```diff
@@ -0,0 +1,36 @@
+sync:
```
Comment (NadiaHG) [inline cob_hardware_config/cob4-5/config/torso_driver.yaml]: remove file

File: cob_hardware_config/cob4-5/config/head_cartesian_controller.yaml (hunk 4 above)
Comment (floweisshardt) [inline cob_hardware_config/cob4-5/config/head_cartesian_controller.yaml:1]: the cob4-5 head will be upgraded later on, so let's keep the config files for the head

File: cob_hardware_config/cob4-5/config/head_controller.yaml (hunk 5 above)
Comment (floweisshardt) [inline cob_hardware_config/cob4-5/config/head_controller.yaml:1]: the cob4-5 head will be upgraded later on, so let's keep the config files for the head

File: cob_hardware_config/cob4-5/config/head_self_collision.yaml (hunk 6 above)
Comment (floweisshardt) [inline cob_hardware_config/cob4-5/config/head_self_collision.yaml:1]: the cob4-5 head will be upgraded later on, so let's keep the config files for the head

**Comment by NadiaHG** [comment]
This pull request requires https://github.com/ipa320/cob_common/pull/192 and https://github.com/ipa320/cob_calibration_data/pull/102

**Comment by NadiaHG** [comment]
Test on cob4-5: OK

**Comment by floweisshardt** [comment]
before merging this, we'll have to wait for the sick_3vistort release to be in the public repo, see http://repositories.ros.org/status_page/ros_indigo_default.html?q=sick_3vistort_driver

**Comment by floweisshardt** [comment]
sick sensor is release and public now, so I'll retrigger travis

**Comment by floweisshardt** [comment]
we need to provide a cob_docker_control substitute in cob_substitute before travis is happy

**Comment by floweisshardt** [comment]
substitution for docking merged, retriggered travis: https://travis-ci.org/ipa320/cob_robots/builds/149988406

**Comment by floweisshardt** [comment]
travis says that sick_3vistort_driver is not available (install tags missing). Will be available soon with new release. Short term fix is to add it as a rosinstall dependency.
Still we have to wait for https://github.com/SICKAG/sick_3vistort/pull/7 to be merged.

**Comment by floweisshardt** [comment]
wow, finally Travis is happy!!!

**Pull request merged**